src/LuaTransformer.ts

```diff
@@ -2685,28 +2701,27 @@ export class LuaTransformer {
             );
         }
 
-        const lhs = this.transformExpression(expression.left);
-        const rhs = this.transformExpression(expression.right);
-
         // Transpile operators
-        const operator = expression.operatorToken.kind;
         switch (operator) {
             case ts.SyntaxKind.AmpersandToken:
             case ts.SyntaxKind.BarToken:
             case ts.SyntaxKind.CaretToken:
             case ts.SyntaxKind.LessThanLessThanToken:
             case ts.SyntaxKind.GreaterThanGreaterThanToken:
             case ts.SyntaxKind.GreaterThanGreaterThanGreaterThanToken:
-                return this.transformBinaryBitOperation(expression, lhs, rhs, operator);
             case ts.SyntaxKind.PlusToken:
             case ts.SyntaxKind.AmpersandAmpersandToken:
             case ts.SyntaxKind.BarBarToken:
             case ts.SyntaxKind.MinusToken:
             case ts.SyntaxKind.AsteriskToken:
             case ts.SyntaxKind.AsteriskAsteriskToken:
             case ts.SyntaxKind.SlashToken:
-            case ts.SyntaxKind.PercentToken:
+            case ts.SyntaxKind.PercentToken: {
+                const lhs = this.transformExpression(expression.left);
```

Comment:
Any reason we calculate these in here at all, instead of inside `transformBinaryOperation`? Seems like a historic thing that's probably not relevant anymore?

Reply:
`transformBinaryOperation` expects already transformed expressions, because it is sometimes given manually constructed ones (like from compound assignments)